redis-persistence/src/main/java/com/netflix/conductor/dao/dynomite/RedisMetadataDAO.java

```diff
@@ -174,11 +179,21 @@ public WorkflowDef getLatest(String name) {
         Preconditions.checkNotNull(name, "WorkflowDef name cannot be null");
         WorkflowDef def = null;
 
-        String workflowDefJsonString = dynoClient.hget(nsKey(WORKFLOW_DEF, name), LATEST);
-        if (workflowDefJsonString != null) {
-            def = readValue(workflowDefJsonString, WorkflowDef.class);
-            recordRedisDaoRequests("getWorkflowDef");
-            recordRedisDaoPayloadSize("getWorkflowDef", workflowDefJsonString.length(), "n/a", def.getName());
+        // Get the list of valid versions leaving Latest
```

Comment:
You can leverage the stream api to get the max version number and get rid of intermediate collection and sorting.

```
 Optional<Integer> optionalMaxVersion = dynoClient.hkeys(nsKey(WORKFLOW_DEF, name)).stream()
                .filter(key -> !key.equals(LATEST))
                .map(Integer::valueOf)
                .max(Comparator.naturalOrder());
``` 

or you can fully compose the function in the stream itself:
```
return dynoClient.hkeys(nsKey(WORKFLOW_DEF, name)).stream()
                .filter(key -> !key.equals(LATEST))
                .map(Integer::valueOf)
                .max(Comparator.naturalOrder())
                .map(String::valueOf)
                .map(workflowVersion -> Optional.ofNullable(dynoClient.hget(nsKey(WORKFLOW_DEF, name), workflowVersion))
                        .map(workflowDefinitionString -> readValue(workflowDefinitionString, WorkflowDef.class))
                        .orElse(null))
                .orElse(null);
```